actionview/test/template/sanitize_helper_test.rb

```diff
@@ -1,62 +1,298 @@
+  test "strip_tags calls sanitize on the full_sanitizer" do
+    assert_equal(
+      "full_sanitizer#sanitize / asdf / {}",
+      self.strip_tags("asdf"),
+    )
+    assert_predicate(self.strip_tags("asdf"), :html_safe?)
   end
 
-  def test_sanitized_allowed_tags_class_method
-    expected = Set.new(["strong", "em", "b", "i", "p", "code", "pre", "tt", "samp", "kbd", "var",
-      "sub", "sup", "dfn", "cite", "big", "small", "address", "hr", "br", "div", "span", "h1", "h2",
-      "h3", "h4", "h5", "h6", "ul", "ol", "li", "dl", "dt", "dd", "abbr", "acronym", "a", "img",
-      "blockquote", "del", "ins"])
-    assert_equal(expected, self.class.sanitized_allowed_tags)
+  test "strip_links calls sanitize on the link_sanitizer" do
+    assert_equal(
+      "link_sanitizer#sanitize / asdf / {}",
+      self.strip_links("asdf"),
+    )
+    assert_not_predicate(self.strip_links("asdf"), :html_safe?)
   end
 
-  def test_sanitized_allowed_attributes_class_method
-    expected = Set.new(["href", "src", "width", "height", "alt", "cite", "datetime", "title", "class", "name", "xml:lang", "abbr"])
-    assert_equal(expected, self.class.sanitized_allowed_attributes)
+  private
+    def new_mock_vendor
+      Class.new do
+        class << self
+          def full_sanitizer
+            new_mock_sanitizer("full_sanitizer")
+          end
+
+          def link_sanitizer
+            new_mock_sanitizer("link_sanitizer")
+          end
+
+          def safe_list_sanitizer
+            new_mock_sanitizer("safe_list_sanitizer").tap do |sanitizer|
+              sanitizer.class_eval do
+                def sanitize_css(style)
+                  "safe_list_sanitizer#sanitize_css / #{style}"
+                end
+              end
+            end
+          end
+
+          def new_mock_sanitizer(injected_name)
+            Class.new do
+              define_method :injected_name do
+                injected_name
+              end
+
+              def sanitize(html, options = {})
+                "#{injected_name}#sanitize / #{html} / #{options}"
+              end
+            end
+          end
+        end
+      end
+    end
+end
+
+module SanitizeHelperVendor
+  #
+  #  Test suite for interactions with each supported sanitizer vendor. Exercise the basic API and
+  #  behavior of the vendor and its sanitizers.
+  #
+  #  We don't want to do exhaustive HTML sanitization testing here. Let's assume it's already being
+  #  done upstream by the vendor.
+  #
+  #  Note that Rails::Html::Sanitizer and Rails::HTML4::Sanitizer are identical vendors (but aren't
+  #  the same class). Eventually we will move away from using Rails::Html (a.k.a Rails::HTML), but
+  #  for now we should make sure everything works as expected by testing it.
+  #
+  vendors = [Rails::Html::Sanitizer, Rails::HTML4::Sanitizer]
+  vendors << Rails::HTML5::Sanitizer if Rails::HTML::Sanitizer.html5_support?
+
+  vendors.each do |vendor|
+    Class.new(ActiveSupport::TestCase) do
```

Comment:
Can we define 3 static classes and add all shared behavior into a module we include into those 3 classes?

Reply:
Done